client/coral-framework/services/postMessage.js

```diff
@@ -0,0 +1,86 @@
+import set from 'lodash/set';
+import has from 'lodash/has';
+import get from 'lodash/get';
+import remove from 'lodash/remove';
+
+/**
+ * createPostMessage returns a service that deals with cross
+ * window communication using the postMessage API.
+ * @param  {Object}  options
+ * @return {Object}  messenger service
+ */
+export function createPostMessage(origin, scope = 'client') {
```

Comment:
At what point do we replace a function that returns an object with a class constructor instead?